.github/workflows/build-all.yml

```diff
@@ -0,0 +1,241 @@
+name: Matrix Build
+
+on:
+  push:
+    branches: [ 'main' ]
+    tags:
+      - 'v*'
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  
+  build_and_test:
+    name: Build and Test Images 
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    strategy:
+      matrix:
+        os_version: [
+          'buster',
+          'ubuntu16',
+          'ubuntu18'
+        ]
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.os_version }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-${{ matrix.os_version }}
+            ${{ runner.os }}-buildx
+
+      - name: Lowercase repo owner
```

Comment:
`Improvement: Could be replaced by this GHA
https://github.com/ASzc/change-string-case-action

Reply:
I wouln't do that that would feeling for me like the lodash approach in CI CD Pipelines.